File: ios/Sources/TaigiKeyboard/Settings/SettingsResetCoordinator.swift
import Foundation
import KeyboardKit

/// Resets settings and user-owned data stores.
///
/// Two surfaces, intentionally separate:
/// - `resetAll()` — pure settings reset (SharedSettings + KeyboardKit defaults).
/// - `resetAllUserData()` — destructive: wipes user frequency, next-word and learned-phrase data.
///
/// `SharedSettings.resetToDefaults()` can't touch the KeyboardKit store
/// directly without importing KeyboardKit in the (soon-to-be engine-only)
/// settings module. The coordinator is the composition point.
enum SettingsResetCoordinator {
    private static let logger = DebugLogger(category: "SettingsResetCoordinator")

    /// Reset the three KeyboardKit-owned user defaults that SharedSettings
    /// does not own. Keep these in sync with `SettingsTab` "reset all" UX.
    static func resetKeyboardKitDefaults() {
        KeyboardSettings.store.set(true, forKey: "com.keyboardkit.settings.keyboard.isAutocapitalizationEnabled")
        KeyboardSettings.store.set(true, forKey: "com.keyboardkit.settings.feedback.isAudioFeedbackEnabled")
        KeyboardSettings.store.set(true, forKey: "com.keyboardkit.settings.feedback.isHapticFeedbackEnabled")
    }

    /// Reset every setting to its default — SharedSettings + KeyboardKit.
    /// Does NOT touch user-owned databases; call `resetAllUserData()` for that.
    static func resetAll() {
        SharedSettings.shared.resetToDefaults()
        resetKeyboardKitDefaults()
    }

    /// Empty user-owned learning data (frequency + next-word association +
    /// learned phrases) in place — the engine's stores (roadmap P7b). The
    /// engine attempts every store even when one fails, so a partial failure
    /// still clears what it can; failures are logged, never thrown.
    static func resetAllUserData() {
        Task {
            do {
                try await CompositionRoot.userData.clearLearningRecords()
            } catch {
                logger.error("Failed to clear learning records: \(error)")
            }
        }
    }
}

File: android/app/src/main/java/com/siansiansu/taigikeyboard/settings/SettingsResetCoordinator.kt
package com.siansiansu.taigikeyboard.settings

import com.siansiansu.taigikeyboard.ime.core.CompositionRoot
import com.siansiansu.taigikeyboard.ime.core.PrefHelper

// Resets settings and user-owned data stores.
//
// Two surfaces, intentionally separate, mirroring
// ios/Sources/TaigiKeyboard/Settings/SettingsResetCoordinator.swift:
// - resetAll(prefs): pure settings reset.
// - resetAllUserData(root): destructive — wipes user frequency, next-word and
//   learned-phrase data.
//
// The engine owns the stores (roadmap P8b) and attempts every one, so a
// partial failure still clears what it can; one that could not be emptied
// surfaces as an exception, reported by the caller's single try/catch.
object SettingsResetCoordinator {
    suspend fun resetAll(prefs: PrefHelper) {
        prefs.resetToDefaults()
    }

    suspend fun resetAllUserData(root: CompositionRoot) {
        root.userData.clearLearningRecords()
    }
}
